flyway-community-db-support/src/main/java/org/flywaydb/community/database/clickhouse/ClickHouseDatabase.java

```diff
@@ -0,0 +1,111 @@
+        return false;
+    }
+
+    @Override
+    public String getBooleanTrue() {
+        return "1";
+    }
+
+    @Override
+    public String getBooleanFalse() {
+        return "0";
+    }
+
+    @Override
+    public boolean catalogIsSchema() {
+        return true;
+    }
+
+    @Override
+    public String getRawCreateScript(Table table, boolean baseline) {
+        String clusterName = configuration.getPluginRegister().getPlugin(ClickHouseConfigurationExtension.class).getClusterName();
+
+        if (StringUtils.hasText(clusterName)) {
+            return "CREATE TABLE IF NOT EXISTS " + table + "_local ON CLUSTER " + clusterName + "(" +
+                    "    installed_rank Int32," +
+                    "    version Nullable(String)," +
+                    "    description String," +
+                    "    type String," +
+                    "    script String," +
+                    "    checksum Nullable(Int32)," +
+                    "    installed_by String," +
+                    "    installed_on DateTime DEFAULT now()," +
+                    "    execution_time Int32," +
+                    "    success Bool" +
+                    ")" +
+                    " ENGINE = ReplicatedMergeTree(" +
+                    "   '/clickhouse/tables/{shard}/" + table.getName() + "'," +
+                    "   '{replica}'" +
+                    " )" +
+                    " PARTITION BY tuple()" +
+                    " ORDER BY (installed_rank);" +
+                    (baseline ? getBaselineStatement(table) + ";" : "") +
+                    "CREATE TABLE IF NOT EXISTS " + table + " ON CLUSTER " + clusterName + "(" +
+                    "    installed_rank Int32," +
+                    "    version Nullable(String)," +
+                    "    description String," +
+                    "    type String," +
+                    "    script String," +
+                    "    checksum Nullable(Int32)," +
+                    "    installed_by String," +
+                    "    installed_on DateTime DEFAULT now()," +
+                    "    execution_time Int32," +
+                    "    success Bool" +
+                    ")" + " ENGINE = Distributed(" + clusterName + ", " + table.getSchema() +
+                    ", " + table.getName() + "_local, 1)";
+        } else {
+            return "CREATE TABLE IF NOT EXISTS " + table + "(" +
+                    "    installed_rank Int32," +
+                    "    version Nullable(String)," +
+                    "    description String," +
+                    "    type String," +
+                    "    script String," +
+                    "    checksum Nullable(Int32)," +
+                    "    installed_by String," +
+                    "    installed_on DateTime DEFAULT now()," +
+                    "    execution_time Int32," +
+                    "    success Bool" +
+                    ")" +
+                    " ENGINE = StripeLog;" +
```

Comment:
On the repair command got: `SQL Error [48] [07000]: Code: 48. DB::Exception: Table engine StripeLog doesn't support mutations. (NOT_IMPLEMENTED) (version 23.2.1.2537 (official build))`

I would like to suggest replacing `ENGINE = StripeLog;` by `ENGINE = MergeTree PRIMARY KEY `version`;`

Reply:
Thanks, fixed that.